source/common/http/conn_manager_impl.cc

```diff
@@ -387,6 +387,14 @@ ConnectionManagerImpl::ActiveStream::ActiveStream(ConnectionManagerImpl& connect
         [this]() -> void { onIdleTimeout(); });
     resetIdleTimer();
   }
+
+  if (connection_manager_.config_.streamRequestTimeout().count()) {
+    request_timeout_ms_ = connection_manager_.config_.streamRequestTimeout();
+    request_timer_ = connection_manager.read_callbacks_->connection().dispatcher().createTimer(
+        [this]() -> void { onRequestTimeout(); });
+    request_timer_->enableTimer(request_timeout_ms_);
```

Comment:
Should the timer be started at construction, or at the first call to encodeHeaders?

Reply:
Yeah I think this is the right place.